tools/src/test/java/org/apache/kafka/tools/GetOffsetShellTest.java

```diff
@@ -391,4 +447,15 @@ private String[] addBootstrapServer(String... args) {
 
         return newArgs.toArray(new String[0]);
     }
+
+    private void retryUntilEqual(List<Row> expected, List<Row> output) {
```

Comment:
the `output` should be a function. Otherwise, the output always returns same value in the waiting loop

Reply:
Ahh yes. I've revised to accept a function now. The output should be refreshed during the wait loop

Reply:
@chia7712 The build still contains falied test even when retry is implemented... looks like it's due to race condition when executing the tests ?

Reply:
> The build still contains falied test even when retry is implemented... looks like it's due to race condition when executing the tests ?

all failed tests are running in kraft mode, so I agree up to a point...